### ISSUE TITLE
[FEATURE] Remove upper bound on Requires-Python

### ISSUE THREAD

**Issue by george-zubrienko** [lidatong/dataclasses-json#452]
### Description

Follow-up from the discussion [here](https://github.com/lidatong/dataclasses-json/pull/449)

### Possible solution

Update Python version constraints to `^3.7` from `>=3.7, <3.13`

### Alternatives

N/A

### Context

[Good morning read](https://discuss.python.org/t/requires-python-upper-limits/12663)

**Comment by george-zubrienko** [comment]
FYI @cdce8p

**Comment by george-zubrienko** [comment]
BTW, this looks a bit alarming to me:

![image](https://github.com/lidatong/dataclasses-json/assets/14901777/5eee4248-3245-48f7-a746-ea610961cd18)

**Comment by cdce8p** [comment]
> BTW, this looks a bit alarming to me - lots of time spend on dependency install

This is at least somewhat to be expected. There are a lot of packages which don't publish wheels for `3.12` just yet. So poetry / pip needs to build those for each dependency install. It could probably be cached though if you wanted to. https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages

**Comment by george-zubrienko** [comment]
> This is at least somewhat to be expected. There are a lot of packages which don't publish wheels for 3.12 just yet. So poetry / pip needs to build those for each dependency install. It could probably be cached though if you wanted to.

Makes sense! I was worried first it was the actual test suite. As a part of this, should we also consider bumping python min to 3.8? Or maybe at least remove test run for 3.7?

**Comment by george-zubrienko** [comment]
Reference download data:

![image](https://github.com/lidatong/dataclasses-json/assets/14901777/bfedea1b-04ca-4e3b-9851-fbd50c870c8b)

**Comment by george-zubrienko** [comment]
![image](https://github.com/lidatong/dataclasses-json/assets/14901777/9b12434b-7ee1-42d6-8558-41cd88ca98e9)

**Comment by george-zubrienko** [comment]
TLDR:
Py3.7 ~25k downloads/day - look state to me
Py3.8-3.10 >50k/day
Py3.11 ~25k /day - growing

https://pypistats.org/packages/dataclasses-json

**Comment by george-zubrienko** [comment]
@lidatong when you have time, your opinion would help a lot here :)
